core/src/conn.rs

```diff
@@ -2,11 +2,12 @@ use crate::odbc_uri::UserOptions;
 use crate::{err::Result, Error};
 use crate::{MongoQuery, TypeMode};
 use bson::{doc, Bson, UuidRepresentation};
-use mongodb::sync::Client;
+use mongodb::Client;
 use serde::{Deserialize, Serialize};
 use std::time::Duration;
 
 #[derive(Debug)]
+#[repr(C)]
```

Comment:
Just for safety, I wanted to ensure the memory layout is preserved across ABI boundary.